src/lib/integrations/web3/index.ts

```diff
@@ -1,23 +1,31 @@
-import typeValidation from "./typeValidation"
+import * as typeValidation from "./typeValidation"
 import * as rawWeb3 from "web3"
 const Web3 = rawWeb3 as any
 
-let web3: any
-if (typeof web3 !== "undefined") {
-  web3 = new Web3(web3.currentProvider)
-} else {
-  // TODO: set the provider you want
-  web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"))
+export const getWeb3 = async () => {
+  const ethereum = (window as any).ethereum
+  const web3 = (window as any).web3
+  if (ethereum) {
+    try {
+      // Request account access if needed
+      await ethereum.enable()
+      // Acccounts now exposed
+      return new Web3(ethereum)
+    } catch (error) {
+      console.log("User denied account access...")
+    }
+  }
+  // Legacy dapp browsers...
+  else if (web3) {
+    return new Web3(web3.currentProvider)
+    // Acccounts always exposed
+  }
+  // Non-dapp browsers...
+  else {
+    console.log(
+      "Non-Ethereum browser detected. You should consider trying MetaMask!"
```

Comment:
Made this bug for improving our error handling in these cases:
* No Support Found
* No RPC Connection
* No Account
* No Network Connection (Privacy Mode)
* Declining Network Connection
* No Funds
* Nonce Mismatch
* Signing Capabilities Unavailable
  * Have the user send a signed message to this app's DAO? Allows network to see user count?

https://github.com/dOrgTech/dOrg/issues/47